tests/unit/test_signature.cpp

```diff
@@ -0,0 +1,427 @@
+/*
+ * #%L
+ * %%
+ * Copyright (C) 2018 BMW Car IT GmbH
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+#include <algorithm>
+#include <fstream>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "IOUtils.h"
+
+#include "signature.cpp"
+
+using namespace std::string_literals;
+
+using namespace mococrw;
+
+class SignatureTest : public ::testing::Test
```

Comment:
Some generic observations:
* Have I missed a use-case where we sign, modify the content and check the signature?
* There is a lot of unit-test which check various age-cases in configuring the context. It basically tests OpenSSL, not our wrapper. Value of such tests is questionable.
* After a brief look, it seems like we have a lot of overlapping tests. I would recommend using test-coverage tooling to reduce the number of tests (if possible).